aspnet-core/src/myvetjob.Application/Jobs/JobAppService.cs

```diff
@@ -13,15 +13,30 @@ public JobAppService(IJobManager jobManager)
         {
             _jobManager = jobManager;
         }
-        public async Task<JobDto> GetUnexpiredJobByIdAsync(int jobId)
+        public async Task<JobDto> GetActiveJobByIdAsync(int jobId)
         {
-            var job = await _jobManager.GetUnexpiredJobByIdAsync(jobId);
+            var job = await _jobManager.GetAsync(jobId);
             return ObjectMapper.Map<JobDto>(job);
         }
-        public async Task<PagedResultDto<JobDto>> GetUnexpiredJobsAsync()
+        /// <summary>
+        /// Retrieves a list of active jobs asynchronously. That is, jobs that have been paid for and have not expired.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<PagedResultDto<JobDto>> GetActiveJobsAsync(GetActiveJobsInput input)
         {
-            var jobs = await _jobManager.GetUnexpiredJobsAsync();
-            var totalJobs = await _jobManager.GetCountUnexpiredJobsAsync();
+            var getAllJobsInput = new GetAllJobsInput
+            {
+                Keyword = input.Keyword,
+                EmploymentType = input.EmploymentType,
+                IncludeExpiredJobs = false,
```

Comment:
Important, since this is meant to be a public endpoint, we map from get active jobs to get all jobs and manually set includeexpiredjobs and orderstatus filters.